conversions/test/Test/Data/Conversions/Integral.hs

```diff
@@ -0,0 +1,265 @@
+module Test.Data.Conversions.Integral (testTree) where
+
+import Data.Conversions
+import Data.Int (Int, Int16, Int32, Int64, Int8)
+import Data.Typeable (Typeable)
+import Data.Word (Word, Word16, Word32, Word64, Word8)
+import GHC.Enum (pred, succ)
+import GHC.Integer (Integer)
+import MPrelude
+
+import qualified Data.Conversions.Hedgehog as Test
+import qualified Hedgehog
+import qualified Hedgehog.Gen
+import qualified Hedgehog.Range
+import qualified Test.Tasty                as Tasty
+
+type BoundResult a b = Either (BoundError a b) b
+type Result a b      = Either (UserBoundError a b) b
+
+testTree :: Tasty.TestTree
+testTree
+  = Tasty.testGroup "Integral"
+    [ integersToNatural
+    , intToWord
+    , word64ToInt
+    , integerToIntegral
+    , naturalToWord
+    ]
+
+integersToNatural :: Tasty.TestTree
+integersToNatural
+  = Tasty.testGroup "Integers to Natural conversions"
+    [ toNatural Hedgehog.Gen.int
+    , toNatural Hedgehog.Gen.int8
+    , toNatural Hedgehog.Gen.int16
+    , toNatural Hedgehog.Gen.int32
+    , toNatural Hedgehog.Gen.int64
+    , integerToNatural
+    ]
+  where
+    toNatural
+      :: forall integer
+      .  ( Bounded integer
+         , Conversion (Result Natural integer) Natural
+         , Conversion (Result integer Natural) integer
+         , Num integer
+         , Show integer
+         , Typeable integer
+         )
+      => (Hedgehog.Range integer -> Hedgehog.Gen integer)
+      -> Tasty.TestTree
+    toNatural genInteger
+      = Test.run biConvert generators
+      where
+        generators :: Test.Generators integer
+        generators = Test.Generators
+          { success = [genInteger range]
+          , failure = [pure $ minBound @integer]
```

Comment:
__suggestion:__ It looks like this is missing the exact boundary check for integer, which would be `-1`.

Reply:
> Just demonstrating that -1 is invalid is probably fine. wdyt?


I agree and using `-1` resonates well with our principle of using the simplest primitive. `-1` is the simplest in this case.